ts/features/payments/transaction/store/selectors/index.ts

```diff
@@ -1,15 +1,98 @@
-import _ from "lodash";
+import _, { values } from "lodash";
 import * as pot from "@pagopa/ts-commons/lib/pot";
 import { createSelector } from "reselect";
+import { IndexedById } from "../../../../../store/helpers/indexer";
+import { isSuccessTransaction, Transaction } from "../../../../../types/pagopa";
+
 import { GlobalState } from "../../../../../store/reducers/types";
-import { latestTransactionsSelector } from "../../../../../store/reducers/wallet/transactions";
+
+/**
+ * The transactions selector will truncate the list at this length
+ */
+const MAX_TRANSACTIONS_IN_LIST = 500;
+
+export type TransactionsState = Readonly<{
+  transactions: pot.Pot<IndexedById<Transaction>, Error>;
+  total: pot.Pot<number, Error>;
+}>;
 
 const walletTransactionSelector = (state: GlobalState) =>
-  state.features.payments.transaction;
+  state.features.payments.legacyTransaction;
 
 export const walletTransactionDetailsPotSelector = (state: GlobalState) =>
   walletTransactionSelector(state).details;
 
+const potTransactions = (state: GlobalState) =>
+  state.features.payments.legacyTransaction.transactions;
+
+// selectors
+export const getTransactions = createSelector(
+  potTransactions,
+  transactionsPot =>
+    pot.map(
+      transactionsPot,
+      txs =>
+        values(txs).filter(value =>
+          isSuccessTransaction(value)
+        ) as ReadonlyArray<Transaction>
+    )
+);
+
+export const latestTransactionsSelector = createSelector(
+  getTransactions,
+  transactionsPot =>
+    pot.map(
+      transactionsPot,
+      transactions =>
+        [...transactions]
+          .sort((a, b) =>
+            // FIXME: code here is checking for NaN assuming creation dates may
+            //        be undefined, but since we override the pagoPA Wallet
+            //        type to force creation dates to always be defined and we
+            //        use that new type for parsing responses, we ignore
+            //        wallets with undefined creation dates... so the check
+            //        is unnecessary.
+
+            isNaN(a.created as any) || isNaN(b.created as any)
```

Comment:
This function isn't new but was inherited from the legacy selector in previous files. Unfortunately, this functionality still needs to exist because it's related to the list of legacy transactions (found by navigating: 'Payments Section -> CTA See All -> Scroll to the end of the list -> CTA See Previous Transactions'). In this list, we have the so-called legacy transactions that must remain in the app until the backend completes the migration from the old to the new system. Once that migration is done, this functionality, along with its methods and selectors, can be removed from the app.
That said I think that we can leave this line of code since the next month will be deleted with its functionality, what do you think?